mm2src/coins/lightning/ln_platform.rs

```diff
@@ -601,15 +601,40 @@ impl BroadcasterInterface for Platform {
         let tx_hex = serialize_hex(tx);
         debug!("Trying to broadcast transaction: {}", tx_hex);
 
-        let fut = self.coin.send_raw_tx(&tx_hex);
+        let platform_coin = self.coin.clone();
         let fut = async move {
-            match fut.compat().await {
-                Ok(id) => info!("Transaction broadcasted successfully: {:?} ", id),
-                // TODO: broadcast transaction through p2p network in case of error
-                Err(e) => error!("Broadcast transaction {} failed: {}", txid, e),
+            loop {
+                let bytes = match hex::decode(&tx_hex) {
+                    Ok(b) => b,
+                    Err(e) => {
+                        error!("Converting transaction to bytes error:{}", e);
+                        break;
+                    },
+                };
```

Comment:
I think you can take this out of the loop, there's no need to repeat it in case of network failure

Reply:
done